packages/web/lib/hooks/useUserXp.ts

```diff
@@ -0,0 +1,97 @@
+import { Constants } from '@metafam/utils';
+import { ethers } from 'ethers';
+import { useEffect, useState } from 'react';
+import { SCAccount, SCAccountsData } from 'sourcecred';
+
+interface XPProps {
+  userTotalXP: number;
+  variationThisWeek: number;
+  variationLastWeek: number;
+  thisWeekXP: number;
+  lastWeekXP: number;
+  userWeeklyCred: number[];
+}
+
+export const useUserXP = (userAddress: string): XPProps => {
+  const [XPDataObj, setXPDataObj] = useState<XPProps>({
+    userTotalXP: 0,
+    variationThisWeek: 0,
+    variationLastWeek: 0,
+    thisWeekXP: 0,
+    lastWeekXP: 0,
+    userWeeklyCred: [],
+  });
```

Comment:
Why isn't this defaulted to `null`? It is possible for `getXP` to return `null` & that seems more concise.

Reply:
Well null won't display anything on the screen and won't look good everytime a user visits the page
<img width="341" alt="Screenshot 2021-12-02 at 8 41 36 PM" src="https://user-images.githubusercontent.com/53316345/144449063-2c5a0061-f7ed-4688-86cb-51108359a24f.png"
<img width="328" alt="Screenshot 2021-12-02 at 8 43 42 PM" src="https://user-images.githubusercontent.com/53316345/144449501-82693dd8-bcbf-4709-9338-5a51db4d7228.png">
>

Reply:
That's not actually correct though. When the component first initializes or on a failure to retrieve the XP (network failure, for example) the value is unknown, not 0.

Rather than 0, you should display something like *Unknown* in the XP field.

Reply:
OK